### PR TITLE
[HttpKernel] Improve PHPDoc on `#[AsController]` attribute

### LINKED ISSUE #51920: Improve PHPDoc descriptions of Attribute classes and parameters
### Description

Attributes are used in application codebase to configure specific behaviors. It's sometimes difficult, especially for a novice, to understand the purpose of an attribute or one of its parameters. To improve the developer experience, I suggest adding a description for each attribute class and parameter. Even if it seems obvious, it's a good idea to be explicit.

If you want to work on a class, or a set of classes, please comment this issue before to avoid duplicate work by someone else.

List of all attributes classes to track the progress (checked = doc is complete):

### Can be ignored -> deprecated
- [x] [`Symfony\Component\DependencyInjection\Attribute\MapDecorated`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/MapDecorated.php)

### Can be ignored -> already good 👍 
- [x] [`Symfony\Component\DependencyInjection\Attribute\Exclude`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/Exclude.php)

### Todo

- [ ] [`Symfony\Component\Validator\Constraints\GroupSequenceProvider`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/GroupSequenceProvider.php)

### In progress

- [ ] [`Symfony\Component\Validator\Constraints\GroupSequence`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/GroupSequence.php) #53428
- [x] [`Symfony\Bundle\FrameworkBundle\Routing\Attribute/AsRoutingConditionService`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Bundle/FrameworkBundle/Routing/Attribute/AsRoutingConditionService.php) #51996
- [ ] [`Symfony\Component\RemoteEvent\Attribute\AsRemoteEventConsumer`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/RemoteEvent/Attribute/AsRemoteEventConsumer.php) #51996
- [ ] [`Symfony\Component\Routing\Annotation\Route`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Routing/Annotation/Route.php) #51996
- [ ] [`Symfony\Component\Scheduler\Attribute\AsCronTask`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Scheduler/Attribute/AsCronTask.php)#51996
- [ ] [`Symfony\Component\Scheduler\Attribute\AsPeriodicTask`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Scheduler/Attribute/AsPeriodicTask.php) #51996
- [ ] [`Symfony\Component\Scheduler\Attribute\AsSchedule`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Scheduler/Attribute/AsSchedule.php) #51996

### Already done

- [x] [`Symfony\Component\Validator\Constraints\Isbn`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Isbn.php) #52012
- [x] [`Symfony\Bridge\Doctrine\Attribute\MapEntity`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php) #51987
- [x] [`Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntity.php) #51987
- [x] [`Symfony\Bridge\Twig\Attribute\Template`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Bridge/Twig/Attribute/Template.php) #51987
- [x] [`Symfony\Component\Console\Attribute\AsCommand`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Console/Attribute/AsCommand.php) #51974
- [x] [`Symfony\Component\DependencyInjection\Attribute\AsAlias`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/AsAlias.php) #51971
- [x] [`Symfony\Component\DependencyInjection\Attribute\AsDecorator`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/AsDecorator.php) #51971
- [x] [`Symfony\Component\DependencyInjection\Attribute\AsTaggedItem`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/AsTaggedItem.php) #51971
- [x] [`Symfony\Component\DependencyInjection\Attribute\Autoconfigure`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/Autoconfigure.php) #51971
- [x] [`Symfony\Component\DependencyInjection\Attribute\AutoconfigureTag`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/AutoconfigureTag.php) #51971
- [x] [`Symfony\Component\DependencyInjection\Attribute\Autowire`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/Autowire.php)
- [x] [`Symfony\Component\DependencyInjection\Attribute\AutowireCallable`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/AutowireCallable.php) #51971
- [x] [`Symfony\Component\DependencyInjection\Attribute\AutowireDecorated`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/AutowireDecorated.php) #51971
- [x] [`Symfony\Component\DependencyInjection\Attribute\AutowireIterator`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/AutowireIterator.php) #51971
- [x] [`Symfony\Component\DependencyInjection\Attribute\AutowireLocator`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/AutowireLocator.php) #51971
- [x] [`Symfony\Component\DependencyInjection\Attribute\AutowireServiceClosure`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/AutowireServiceClosure.php) #51971
- [x] [`Symfony\Component\DependencyInjection\Attribute\TaggedIterator`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/TaggedIterator.php) #51971
- [x] [`Symfony\Component\DependencyInjection\Attribute\TaggedLocator`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/TaggedLocator.php) #51971
- [x] [`Symfony\Component\DependencyInjection\Attribute\Target`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/Target.php) #51971
- [x] [`Symfony\Component\DependencyInjection\Attribute\When`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Attribute/When.php) #51971
- [x] [`Symfony\Component\EventDispatcher\Attribute\AsEventListener`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/EventDispatcher/Attribute/AsEventListener.php) #51974
- [x] [`Symfony\Component\HttpKernel\Attribute\AsController`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpKernel/Attribute/AsController.php) #52147
- [x] [`Symfony\Component\HttpKernel\Attribute\AsTargetedValueResolver`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpKernel/Attribute/AsTargetedValueResolver.php) #51971
- [x] [`Symfony\Component\HttpKernel\Attribute\Cache`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpKernel/Attribute/Cache.php) #51968
- [x] [`Symfony\Component\HttpKernel\Attribute\MapDateTime`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpKernel/Attribute/MapDateTime.php) #51971
- [x] [`Symfony\Component\HttpKernel\Attribute\MapQueryParameter`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpKernel/Attribute/MapQueryParameter.php) #51971
- [x] [`Symfony\Component\HttpKernel\Attribute\MapQueryString`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpKernel/Attribute/MapQueryString.php) #51971
- [x] [`Symfony\Component\HttpKernel\Attribute\MapRequestPayload`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpKernel/Attribute/MapRequestPayload.php) #51971
- [x] [`Symfony\Component\HttpKernel\Attribute\ValueResolver`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpKernel/Attribute/ValueResolver.php) #51971
- [x] [`Symfony\Component\HttpKernel\Attribute\WithHttpStatus`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpKernel/Attribute/WithHttpStatus.php) #51971
- [x] [`Symfony\Component\HttpKernel\Attribute\WithLogLevel`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpKernel/Attribute/WithLogLevel.php) #51971
- [x] [`Symfony\Component\Messenger\Attribute\AsMessageHandler`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Messenger/Attribute/AsMessageHandler.php) #52475
- [x] [`Symfony\Component\Security\Core\Validator/Constraints/UserPassword`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Security/Core/Validator/Constraints/UserPassword.php) #51974
- [x] [`Symfony\Component\Security\Http\Attribute/CurrentUser`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Security/Http/Attribute/CurrentUser.php) #51974
- [x] [`Symfony\Component\Security\Http\Attribute/IsGranted`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Security/Http/Attribute/IsGranted.php) #51974
- [x] [`Symfony\Component\Serializer\Annotation\Context`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Serializer/Annotation/Context.php) #51974
- [x] [`Symfony\Component\Serializer\Annotation\DiscriminatorMap`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Serializer/Annotation/DiscriminatorMap.php) #51974
- [x] [`Symfony\Component\Serializer\Annotation\Groups`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Serializer/Annotation/Groups.php) #51974
- [x] [`Symfony\Component\Serializer\Annotation\Ignore`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Serializer/Annotation/Ignore.php) #51974
- [x] [`Symfony\Component\Serializer\Annotation\MaxDepth`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Serializer/Annotation/MaxDepth.php) #51974
- [x] [`Symfony\Component\Serializer\Annotation\SerializedName`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Serializer/Annotation/SerializedName.php) #51974
- [x] [`Symfony\Component\Serializer\Annotation\SerializedPath`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Serializer/Annotation/SerializedPath.php) #51974
- [x] [`Symfony\Component\Validator\Attribute\HasNamedArguments`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Attribute/HasNamedArguments.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\All`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/All.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\AtLeastOneOf`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/AtLeastOneOf.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Bic`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Bic.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Blank`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Blank.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Callback`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Callback.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\CardScheme`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/CardScheme.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Cascade`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Cascade.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Choice`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Choice.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Cidr`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Cidr.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Collection`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Collection.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Count`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Count.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Country`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Country.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\CssColor`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/CssColor.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Currency`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Currency.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Date`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Date.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\DateTime`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/DateTime.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\DisableAutoMapping`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/DisableAutoMapping.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\DivisibleBy`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/DivisibleBy.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Email`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Email.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\EnableAutoMapping`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/EnableAutoMapping.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\EqualTo`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/EqualTo.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Expression`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Expression.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\ExpressionLanguageSyntax`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/ExpressionLanguageSyntax.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\ExpressionSyntax`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/ExpressionSyntax.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\File`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/File.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\GreaterThan`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/GreaterThan.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\GreaterThanOrEqual`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/GreaterThanOrEqual.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Hostname`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Hostname.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Iban`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Iban.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\IdenticalTo`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/IdenticalTo.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Image`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Image.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Ip`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Ip.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\IsFalse`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/IsFalse.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Isin`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Isin.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\IsNull`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/IsNull.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Issn`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Issn.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\IsTrue`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/IsTrue.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Json`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Json.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Language`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Language.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Length`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Length.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\LessThan`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/LessThan.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\LessThanOrEqual`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/LessThanOrEqual.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Locale`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Locale.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Luhn`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Luhn.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Negative`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Negative.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\NegativeOrZero`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/NegativeOrZero.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\NoSuspiciousCharacters`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/NoSuspiciousCharacters.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\NotBlank`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/NotBlank.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\NotCompromisedPassword`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/NotCompromisedPassword.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\NotEqualTo`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/NotEqualTo.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\NotIdenticalTo`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/NotIdenticalTo.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\NotNull`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/NotNull.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\PasswordStrength`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/PasswordStrength.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Positive`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Positive.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\PositiveOrZero`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/PositiveOrZero.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Range`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Range.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Regex`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Regex.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Sequentially`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Sequentially.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Time`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Time.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Timezone`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Timezone.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Traverse`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Traverse.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Type`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Type.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Ulid`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Ulid.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Unique`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Unique.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Url`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Url.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Uuid`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Uuid.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\Valid`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/Valid.php) #52012
- [x] [`Symfony\Component\Validator\Constraints\When`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Validator/Constraints/When.php) #52012
- [x] [`Symfony\Component\Workflow\Attribute\AsAnnounceListener`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Workflow/Attribute/AsAnnounceListener.php) #51974
- [x] [`Symfony\Component\Workflow\Attribute\AsCompletedListener`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Workflow/Attribute/AsCompletedListener.php) #51974
- [x] [`Symfony\Component\Workflow\Attribute\AsEnteredListener`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Workflow/Attribute/AsEnteredListener.php) #51974
- [x] [`Symfony\Component\Workflow\Attribute\AsEnterListener`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Workflow/Attribute/AsEnterListener.php) #51974
- [x] [`Symfony\Component\Workflow\Attribute\AsGuardListener`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Workflow/Attribute/AsGuardListener.php) #51974
- [x] [`Symfony\Component\Workflow\Attribute\AsLeaveListener`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Workflow/Attribute/AsLeaveListener.php) #51974
- [x] [`Symfony\Component\Workflow\Attribute\AsTransitionListener`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Workflow/Attribute/AsTransitionListener.php) #51974

### Example

The [`Symfony\Component\HttpKernel\Attribute\Cache`](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpKernel/Attribute/Cache.php) class is the perfect example of what needs to be done:

- Add a description to the class header to indicate its purpose and what it does
- Add a description of each constructor argument

### PR DESCRIPTION
**Pull request by muchafm**
| Q             | A
| ------------- | ---
| Branch?       | 6.4 
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #51920
| License       | MIT

Improve PHPDoc on AsController attribute

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Attribute/AsController.php [hunk 1]
```diff
@@ -12,7 +12,10 @@
 namespace Symfony\Component\HttpKernel\Attribute;
 
 /**
- * Service tag to autoconfigure controllers.
+ * To autoconfigure your controllers as serivces you can use this PHP attribute.
+ * This attribute automatically tags your controller as `controller.service_arguments`.
+ * 
+ * See more at https://symfony.com/doc/current/controller/service.html
```
Comment (BafS) [inline src/Symfony/Component/HttpKernel/Attribute/AsController.php]:
```suggestion
 * @see https://symfony.com/doc/current/controller/service.html
```

File: src/Symfony/Component/HttpKernel/Attribute/AsController.php [hunk 2]
```diff
@@ -12,7 +12,11 @@
 namespace Symfony\Component\HttpKernel\Attribute;
 
 /**
- * Service tag to autoconfigure controllers.
+ * To autoconfigure your controllers as serivces you can use
```
Comment (javiereguiluz) [inline src/Symfony/Component/HttpKernel/Attribute/AsController.php]: Thanks @muchafm. I like the improvement ... but I'd reword it and expand it a bit. For example:

```
 * Autoconfigures controllers as services by applying the `controller.service_arguments`
 * tag to them. This allows to inject services in controller action methods, not
 * only in their constructors.
 *
 * @see https://symfony.com/doc/current/controller/service.html
```

Comment (muchafm) [reply src/Symfony/Component/HttpKernel/Attribute/AsController.php]: Oh I see, thank you for the suggestion, I will apply it

**Review by GromNaN — APPROVED** [review]

File: src/Symfony/Component/HttpKernel/Attribute/AsController.php [hunk 3]
```diff
@@ -12,7 +12,11 @@
 namespace Symfony\Component\HttpKernel\Attribute;
 
 /**
- * Service tag to autoconfigure controllers.
+ * Autoconfigures controllers as services by applying the `controller.service_arguments`
+ * tag to them. This allows to inject services in controller action methods, not
+ * only in their constructors.
+ * 
+ * @see https://symfony.com/doc/current/controller/service.html
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Attribute/AsController.php]: WDYT of this? (note that we don't link to the doc usually).

```suggestion
 * tag to them. This enables injecting services as method arguments
 * in addition to other conventional dependency injection strategies.
```

Comment (muchafm) [reply src/Symfony/Component/HttpKernel/Attribute/AsController.php]: It is better I think, thank you for your feedback !

**Comment by nicolas-grekas** [comment]
Thank you @muchafm.
